pkg/api/types.go

```diff
@@ -739,6 +739,9 @@ type SecretVolumeSource struct {
 	// mode, like fsGroup, and the result can be other mode bits set.
 	// +optional
 	DefaultMode *int32
+	// Specify whether the Secret must be defined
+	// +optional
+	Optional *bool
```

Comment:
Should the optional-ness apply to keys inside secrets/configmaps?  e.g. The CM exists but the key specified does not.